neo/rawio/neuralynxrawio.py

```diff
@@ -672,23 +910,33 @@ def read_txt_header(filename):
     if old_date_format:
         datetime1_regex = r'## Time Opened \(m/d/y\): (?P<date>\S+)  \(h:m:s\.ms\) (?P<time>\S+)'
         datetime2_regex = r'## Time Closed \(m/d/y\): (?P<date>\S+)  \(h:m:s\.ms\) (?P<time>\S+)'
-        filename_regex = r'## File Name (?P<filename>\S+)'
+        # filename_regex = r'## File Name (?P<filename>\S+)'  # not used
         datetimeformat = '%m/%d/%Y %H:%M:%S.%f'
     else:
         datetime1_regex = r'-TimeCreated (?P<date>\S+) (?P<time>\S+)'
         datetime2_regex = r'-TimeClosed (?P<date>\S+) (?P<time>\S+)'
-        filename_regex = r'-OriginalFileName "?(?P<filename>\S+)"?'
+        # filename_regex = r'-OriginalFileName "?(?P<filename>\S+)"?'  # not used
         datetimeformat = '%Y/%m/%d %H:%M:%S'
 
-    original_filename = re.search(filename_regex, txt_header).groupdict()['filename']
+    # original_filename = re.search(filename_regex, txt_header).groupdict()['filename']  Not used
+
+    dt1 = re.search(datetime1_regex, txt_header)  # cant call groupdict if re.search fails
+    dt2 = re.search(datetime2_regex, txt_header)
 
-    dt1 = re.search(datetime1_regex, txt_header).groupdict()
-    dt2 = re.search(datetime2_regex, txt_header).groupdict()
+    # Using this at least the code won't break if the regexp fails
+    if dt1 is not None:
+        dt1 = dt1.groupdict()
+        info['recording_opened'] = datetime.datetime.strptime(
+            dt1['date'] + ' ' + dt1['time'], datetimeformat)
+    else:
+        info['recording_opened'] = None
```

Comment:
good idea

Reply:
This is an issue that could be addressed in a more general fashion in #851. @PeterNSteinmetz, what do you think?

Reply:
Yes, @JuliaSprenger, I think this is dealt with in #851, with a different pattern in the NlxHeader parsing code.

Reply:
@PeterNSteinmetz nice update on the io, I think we had a few recording sessions where the recording software was closed too early. In this case there is no closing time in the header (but the keyword is there [example file](https://gin.g-node.org/TRuikes/ephy_testing_data/src/master/neuralynx/examples_breaking_data/dsp_compensation_and_ringbuff_errors))

I fixed it with [these lines](https://github.com/TRuikes/python-neo/blob/3a9d537bb38510bf8863d75b3c08f5fcc29a549f/neo/rawio/neuralynxrawio/nlxheader.py#L217-L224), for our data.

Reply:
@TRuikes I agree that is the right way to handle it and it is reasonable to allow the closing time to be optional generally. Would you create a pull request from that change?